src/rred_reports/dataclass_function.py

```diff
@@ -0,0 +1,189 @@
+    assessi_scotest3: Data[float]
+    assessii_engcheck1: Data[float]
+    assessii_engtest4: Data[float]
+    assessii_engtest5: Data[float]
+    assessii_engtest6: Data[float]
+    assessii_engtest7: Data[float]
+    assessii_engtest8: Data[float]
+    assessii_scotest4: Data[float]
+    assessii_iretest2: Data[float]
+    assessii_iretype2: Data[float]
+    assessiii_engtest10: Data[float]
+    assessiii_engtest11: Data[float]
+    assessiii_engtest9: Data[float]
+    assessiii_iretest4: Data[float]
+    entry_dob: Data[Literal["datetime64[ns]"]]
+    summer: Data[str]
+    entry_date: Data[Literal["datetime64[ns]"]]
+    entry_testdate: Data[Literal["datetime64[ns]"]]
+    exit_date: Data[Literal["datetime64[ns]"]]
+    exit_outcome: Data[str]
+    entry_year: Data[str]
+    entry_gender: Data[str]
+    entry_ethnicity: Data[str]
+    entry_language: Data[str]
+    entry_poverty: Data[str]
+    entry_sen_status: Data[str]
+    entry_special_cohort: Data[str]
+    entry_bl_result: Data[float]
+    entry_li_result: Data[float]
+    entry_cap_result: Data[float]
+    entry_wt_result: Data[float]
+    entry_wv_result: Data[float]
+    entry_hrsw_result: Data[float]
+    entry_bas_result: Data[float]
+    exit_num_weeks: Data[float]
+    exit_num_lessons: Data[float]
+    exit_lessons_missed_ca: Data[float]
+    exit_lessons_missed_cu: Data[float]
+    exit_lessons_missed_ta: Data[float]
+    exit_lessons_missed_tu: Data[float]
+    exit_bl_result: Data[float]
+    exit_li_result: Data[float]
+    exit_cap_result: Data[float]
+    exit_wt_result: Data[float]
+    exit_wv_result: Data[float]
+    exit_hrsw_result: Data[float]
+    exit_bas_result: Data[float]
+    month3_testdate: Data[Literal["datetime64[ns]"]]
+    month3_bl_result: Data[float]
+    month3_wv_result: Data[float]
+    month3_bas_result: Data[float]
+    month6_testdate: Data[Literal["datetime64[ns]"]]
+    month6_bl_result: Data[float]
+    month6_wv_result: Data[float]
+    month6_bas_result: Data[float]
+    rred_qc_parameters: Data[str]
+
+
+@dataclass
+class Teacher(AsFrame):
+    """Teacher information, can link with Pupil df with rreduserID"""
+
+    rred_user_id: Data[str]
+    reg_rr_title: Data[str]
+    school_id: Data[str]
+
+
+@dataclass
+class School(AsFrame):
+    """School information can link with Teacher df with school_id"""
+
+    school_id: Data[str]
+    rrcp_school: Data[str]
+    rrcp_area: Data[int]
+    rrcp_country: Data[int]
+
+
+def createdf(file: Path) -> pd.DataFrame:
+    """Create a nested dataframe from an excel file
+
+    Args:
+        file (Path): Path object pointing to excel file
+
+    Returns:
+        pd.DataFrame: Nested pandas DataFrame
+    """
+    full_data = pd.read_excel(file)
+
+    def clmnlist(i):
+        return list(full_data.iloc[:, i])
+
+    all_schools_df = School(clmnlist(6), clmnlist(5), clmnlist(4), clmnlist(3))
+    all_schools_df = all_schools_df.drop_duplicates()  # pylint: disable=E1101

```

Comment:
Instead of having pylint disabling maybe we should edit our config for pylint. Looks like we can do this [for pandas](https://stackoverflow.com/questions/33961756/disabling-pylint-no-member-e1101-error-for-specific-libraries#35106735)  I think we should be able to do this in the `pyproject.toml` file, under its  `[tool.pylint]` section